app/controllers/questions_controller.rb

```diff
@@ -35,17 +40,25 @@ def update
   end
 
   def destroy
-    if @question.user_id != @current_user.id
-      return render json: { message: 'Not authorized to delete the question' },
-                    status: :unauthorized
-    end
+    return render json: { message: @message }, status: :unauthorized if @message.present?
 
     @question.destroy
     render status: :no_content
   end
 
   private
 
+  def set_message
+    return unless @question.user_id != @current_user.id
+
+    @message = case action_name
+               when 'update'
+                 'Not authorized to update the question'
+               when 'destroy'
+                 'Not authorized to delete the question'
+               end
+  end
```

Comment:
This approach is not good because the way the code is written is confusing. The function "set_message" sets the message on error, and even handles the case if the question has been created by user or not, which is introducing ambiguity. 

The better name of the function can be "validate_owner", then create a message based on the action, and render error if the resource has not been created by the current_user.